src/api/report/reportRouter.ts

```diff
@@ -0,0 +1,45 @@
+import { OpenAPIRegistry } from '@asteasolutions/zod-to-openapi';
+import express, { Request, Response, Router } from 'express';
+import { z } from 'zod';
+
+import { createApiResponse } from '@/api-docs/openAPIResponseBuilders';
+import { handleServiceResponse } from '@/common/utils/httpHandlers';
+
+import { GetReportGenSchema, GetReportSchema, ReportSchema } from './reportModel';
+import { reportService } from './reportService';
+
+export const reportRegistry = new OpenAPIRegistry();
+
+reportRegistry.register('Report', ReportSchema);
+
+export const ReportRouter: Router = (() => {
+  const router = express.Router();
+
+  reportRegistry.registerPath({
+    method: 'get',
+    path: '/report/{verifierAddress}',
+    tags: ['Report'],
+    request: { params: GetReportSchema.shape.params },
+    responses: createApiResponse(z.array(ReportSchema), 'Success'),
+  });
+
+  reportRegistry.registerPath({
+    method: 'get',
+    path: '/report/{verifierId}/{timestamp}/report.md',
+    tags: ['Report'],
+    request: { params: GetReportGenSchema.shape.params },
+    responses: createApiResponse(z.array(ReportSchema), 'Success'),
+  });
+
+  router.get('/:verifierId/:timestamp/report.md', async (_req: Request, res: Response) => {
```

Comment:
Can we only support this in NODE_ENV=development or smth? In production we serve this via nginx either way.